### DISCUSSION TITLE
Extend membership function for usage for association membership management

### DISCUSSION THREAD

**Discussion by Lurkars** [pretix/pretix#2287] [Ideas]
### duration option: end of year
For handling for example association memberships where membership ends with calendar year a duration in month and days does not work. So something like: end of year, end of month etc could also be useful or a specific date.

### automatic ordering before expiring
Also an option to automatically create a new order for extending the membership xx days before expiry.

Currently I am using Pretix to handle association memberships and implement those feature via third party handling this stuff via API (so not using membership feature at all). After announcing the membership feature I had to note that it does not fulfill my use case. Maybe more people than me want to use Pretix for handling membership fees and general membership management in Pretix. So the above stuff is only my minimal requirements, of course one can think of more management stuff like handling membership IDs etc.

**Comment by raphaelm** [comment]
> duration option: end of year

You can technically already do this by setting up an event series with dates from Jan 1st to Dec 31st each. Then you configure membership timeframes to be defined by the event timeframe. Certainly not the easiest way, but I'm not sure it's worth adding special options in there since this will make the configuration significantly more bloated.

> automatic ordering before expiring

This will likely not happen. Recurring payments are something that we want to avoid for the forseeable future, and it's certainly out of scope for pretix to become a full assiciation membership administration tool. The idea behind the membership feature in pretix is more along the lines of "use a proper membership administration tool like jVerein or [byro](https://github.com/byro/byro), then sync membership status to pretix via the API to provide members-only privileges"

**Reply by Lurkars** [reply]
Okay. Thanks for your answer. Than I will stick to handling membership via third party and API.
This then leads only one other feature request from my side: Can you please include the `expires` field in the "Creating Order" endpoint? That would be awesome.
